app/javascript/packs/verify-personal-key.tsx

```diff
@@ -0,0 +1,24 @@
+import { render } from 'react-dom';
+import { FormSteps } from '@18f/identity-form-steps';
+import { personalKeyStep, personalKeyConfirmStep } from '@18f/identity-verify-flow';
+
+interface AppRootValues {
+  personalKey: string;
+}
+
+interface AppRootElement extends HTMLElement {
+  dataset: DOMStringMap & AppRootValues;
+}
+
+const appRoot = document.getElementById('app-root') as AppRootElement;
+const appForm = document.getElementById('app-form') as HTMLFormElement;
+
+render(
+  <FormSteps
+    steps={[personalKeyStep, personalKeyConfirmStep]}
+    initialValues={{ personalKey: appRoot.dataset.personalKey }}
+    promptOnNavigate={false}
+    onComplete={() => appForm.submit()}
```

Comment:
silly question, how do the values get in to this form before we submit it? it looks like the form lives outside of `appRoot`?

Reply:
Not a silly question. The server-side controller doesn't actually care about any submitted values 🙃 The personal key is generated when the page is shown, and the form submission is basically just a redirect.

Page show:

https://github.com/18F/identity-idp/blob/8cb2802f3f23afd1d30689a3998cd06846c5485f/app/controllers/accounts/personal_keys_controller.rb#L13-L22

Page submit:

https://github.com/18F/identity-idp/blob/8cb2802f3f23afd1d30689a3998cd06846c5485f/app/controllers/users/personal_keys_controller.rb#L22-L25

Reply:
🤦 thanks, I forgot that checking the personal key was all JS before anyways